jni/include/faiss_wrapper.h

```diff
@@ -45,17 +45,27 @@ namespace knn_jni {
         // Sets the sharedIndexState for an index
         void SetSharedIndexState(jlong indexPointerJ, jlong shareIndexStatePointerJ);
 
-        // Execute a query against the index located in memory at indexPointerJ.
-        //
-        // Return an array of KNNQueryResults
+         /**
+         *  Execute a query against the index located in memory at indexPointerJ
+         *  
+         * Parameters:
+         * queryEfSearch: -1 indicates to use efsearch value used during index setting 
+         * 
+         * Return an array of KNNQueryResults
+        */
         jobjectArray QueryIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jlong indexPointerJ,
-                                jfloatArray queryVectorJ, jint kJ, jintArray parentIdsJ);
-
-        // Execute a query against the index located in memory at indexPointerJ along with Filters
-        //
-        // Return an array of KNNQueryResults
+                                jfloatArray queryVectorJ, jint kJ, jint queryEfSearchJ, jintArray parentIdsJ);
+
+        /**
+         *  Execute a query against the index located in memory at indexPointerJ along with Filters
+         *  
+         * Parameters:
+         * queryEfSearch: -1 indicates to use efsearch value used during index setting 
+         * 
+         * Return an array of KNNQueryResults
+        */
         jobjectArray QueryIndex_WithFilter(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jlong indexPointerJ,
```

Comment:
Can we create Overloaded method for this with new parameter , Just to Make Sure ...It works with and without EfSearch passed with Query during whole deployment Process and till it goes out.